src/ruby/yast/builtins.rb

```diff
@@ -674,6 +674,49 @@ def self.y2usernote *args
       return nil
     end
 
+    # Wrapper for the C++ version of strftime. Needed because it honors the
+    # system locale settings, something that is not granted by the Ruby version
+    # of strftime.
+    # @see http://www.cplusplus.com/reference/ctime/strftime/
+    # @param time [Time/DateTime] time to format
+    # @param format [String] format string according to C++ strftime
+    #    specification
+    # @return [String] formatted string
+    def self.stftime(time, format)
+      strftime_wrapper(time_to_tm(time), format)
+    end
+
+    # Creates a hash with the same fields that the C++ time structure.
+    # @private used internally by .strftime
+    # @see http://www.cplusplus.com/reference/ctime/tm/
+    # @param time [Time/DateTime] time to convert
+    # @return [Hash] hash with symbols as keys
+    def self.time_to_tm(time)
```

Comment:
OK, you have my approval to break consistency and do it properly :). It should be both, as I am not sure if yardoc recognize private_class_method and mark it properly.